graylog2-web-interface/src/components/systemjobs/SystemJob.jsx

```diff
@@ -3,17 +3,31 @@ import { ProgressBar } from 'react-bootstrap';
 
 import { LinkToNode, Timestamp } from 'components/common';
 
+import ActionsProvider from 'injection/ActionsProvider';
+const SystemJobsActions = ActionsProvider.getActions('SystemJobs');
+
 const SystemJob = React.createClass({
+  _onCancel(job) {
+    return (e) => {
+      e.preventDefault();
+      if (window.confirm(`Are you sure you want to cancel system job "${job.info}"?`)) {
+        SystemJobsActions.cancelJob(job.id);
+      }
+    };
+  },
   render() {
     const job = this.props.job;
     const progress = job.percent_complete < 100 ?
       (<ProgressBar bsStyle="info" active now={job.percent_complete}/>) : <span className="label label-success finished">Finished!</span>;
+    const cancel = job.is_cancelable ? (<span> - <a href="" onClick={this._onCancel(job)}>cancel job</a></span>) : null;
```

Comment:
To be honest I don't have a strong opinion about it. I like the button because it's more consistent with other elements of the interface, but the spacing and placement looks odd imho.

I was looking a bit and I think we can style the button to go to the right, so it looks a bit less weird. I can take care of that in my task of styling things a bit :)